### PR TITLE
Add kms key to available resources

### PR DESCRIPTION
**Pull request by johkerquux**

### CODE REVIEW

File: modules/aws_efs/variables.tf [hunk 1]
```diff
@@ -6,3 +6,9 @@ variable "tags" {
   type    = map(string)
   default = {}
 }
+
+variable "kms-key-arn" {
```
Comment (SohelSaiyed) [inline modules/aws_efs/variables.tf:10]: This variable does not seem to be optional, as I tried without declaring and it throws error:
`Error: "encryption_config.0.provider.0.key_arn": required field is not set`. if that is required variable can you please define in terraform_template.tfvars for other user to set all required values.

File: examples/eks/cluster.tf [hunk 2]
```diff
@@ -18,14 +19,20 @@ module "aae-cluster" {
...
+    }
+  ]
   subnets                     = module.vpc.public_subnets
   vpc_id                      = module.vpc.vpc_id
   tags                        = var.tags
   map_accounts                = var.paas-eks-accounts
   map_roles                   = var.paas-eks-roles
   config_output_path          = ".terraform/kubeconfig"
   workers_additional_policies = var.workers_additional_policies
+  #TODO Variablize these values
```
Comment (ffazzini) [inline examples/eks/cluster.tf:35]: better to add a ticket number as part of the todo comment, so we can track the improvement that we want to apply

File: examples/eks/locals.tf [hunk 3]
```diff
@@ -3,12 +3,14 @@ locals {
   cluster_domain               = "${var.cluster_name}.${var.zone_domain}"
   cluster_url                  = "${local.protocol}://${local.cluster_host}"
   protocol                     = var.http ? "http" : "https"
-  identity_url                 = "${local.protocol}://${var.identity_host}"
+  identity_url                 = "${local.protocol}://${var.acs_host}"
```
Comment (ffazzini) [inline examples/eks/locals.tf:6]: i was expecting to find an `identity_host` on the right side,
what about if identity and acs need to be hosted on different domains?
Is a use case that won't occurr?

File: examples/eks/main.tf [hunk 4]
```diff
@@ -80,7 +94,9 @@ module "alfresco-process-services" {
   kubernetes_token             = data.kubernetes_secret.alfresco-deployment-service-secret.data.token
   kubernetes_api_server        = data.aws_eks_cluster.aae-cluster.endpoint
   enable_acs                   = var.enable_acs
-  identity_url                 = var.create_identity ? "" : var.identity_host
+  acs_url                      = "https://${var.acs_host}"
+  identity_url                 = "https://${var.acs_host}/auth"
```
Comment (ffazzini) [inline examples/eks/main.tf:98]: same as comment above, but if acs and identity are on the same domain will always work fine
if we will need a more modular approach we can make this flexible later on?
if so feel free to resolve this discussion

File: examples/eks/locals.tf (hunk 3 above)
Comment (ffazzini) [reply examples/eks/locals.tf:6]: after reading the whole pr i found an identity host here https://github.com/Alfresco/terraform-alfresco-process/pull/68/files#diff-c9ac8098c5ea9d3e6a9a596ff0c512a4R166

File: examples/eks/cluster.tf (hunk 2 above)
Comment (johkerquux) [reply examples/eks/cluster.tf:35]: we don't have tickets for this, but will have a full code review in the next sprint, this is just to highlight this part

**Pull request merged**